MetaMask/metamask-extension
# Display large and small numbers as decimals instead of scientific notation on token allowance confirmation screens

## Explanation

Very large numbers are shown as decimals instead of scientific notation.
Very small numbers are also shown as decimals instead of scientific notation.

* Fixes #16427

## Screenshots/Screencaps

### Before

https://user-images.githubusercontent.com/63151811/204797905-0e2a0d90-8c3a-430b-9a67-e09416a82927.mov

https://user-images.githubusercontent.com/63151811/204799265-abeee06c-3b86-4a09-8d19-ca9cbbef96a2.mov

### After

https://user-images.githubusercontent.com/63151811/204801910-57811c8d-7694-44f0-b803-4c425ad84ed8.mov

## Manual Testing Steps

### Etherscan
1. Unlock MetaMask
2. Go to `Settings -> Experimental`
3. Turn `ON` toggle button for `Improved token allowance experience`
4. Go to https://etherscan.io/token/0x6b175474e89094c44da98b954eedeac495271d0f#writeContract
5. Connect your wallet
6. Under approve input any address and a number like `98768451321654687654` or `98768451321` or any other number 
7. Click on write
8. On the MetaMask popup, click on `Use default` or paste in that same number in the spending cap field
9. Notice how the number is displayed correctly.

### Test-Dapp
1. Unlock MetaMask
2. Go to `Settings -> Experimental`
3. Turn `ON` toggle button for `Improved token allowance experience`
4. Open the test dapp https://metamask.github.io/test-dapp/
5. Connect your wallet (click on `Connect` button in `Basic Actions` section)
6. Under the `Send Tokens` section, click on `Create Token`
7. Wait for transaction to be confirmed and click on `Approve Tokens`
or
8. Wait for transaction to be confirmed and click on `Approve Tokens Without Gas`
9. Wait for `Token Allowance` screen is displayed and rendered.
10. Enter a number, something like `98768451321654687654` or `98768451321` or any other number, and notice how the number is displayed correctly.
